doc/src/devdocs/types.md

```diff
@@ -82,7 +82,7 @@ f3(A::Array{T}) where {T<:Any} = 3
 f4(A::Array{Any}) = 4
 ```
 
-The signature of `f3` is a `UnionAll` type wrapping a tuple type.
+The signature of `f3` is a `UnionAll` type wrapping an array type.
```

Comment:
Instead we should clarify that the tuple type we mean is `Tuple{typeof(f3), Array{T}} where T`; it's referring to the whole signature.

Reply:
Interesting. That's better, indeed. 
A link to [function calls](https://docs.julialang.org/en/v1/devdocs/functions/#Function-calls-1) that describes the signature tuple stuff would be nice, like:
```The signature - as decribed in [Function calls](@ref) - of `f3` is a `UnionAll` type wrapping a tuple type: `Tuple{typeof(f3), Array{T}} where T`.```

But the docs build fails.
```
git checkout origin/patch-1
make clean
# without a first make, make docs was slow as hell
make -j $(nproc)
make docs

...
┌ Warning: no doc found for reference '[`Pkg.clone()`](@ref)' in `src/manual/packages.md`.
└ @ Documenter.CrossReferences ~/share/prog/julia/dev/julia-dev/doc/deps/packages/Documenter/zOHWz/src/CrossReferences.jl:161
[ Info: CheckDocument: running document checks.
[ Info: Populate: populating indices.
ERROR: LoadError: `makedocs` encountered an error. Terminating build
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] runner(::Type{Documenter.Builder.RenderDocument}, ::Documenter.Documents.Document) at /home/ederag/share/prog/julia/dev/julia-dev/doc/deps/packages/Documenter/zOHWz/src/Builder.jl:204
 [3] dispatch(::Type{Documenter.Builder.DocumentPipeline}, ::Documenter.Documents.Document) at /home/ederag/share/prog/julia/dev/julia-dev/doc/deps/packages/Documenter/zOHWz/src/Utilities/Selectors.jl:167
 [4] (::getfield(Documenter, Symbol("##3#8")))() at /home/ederag/share/prog/julia/dev/julia-dev/doc/deps/packages/Documenter/zOHWz/src/Documenter.jl:267
 [5] cd(::getfield(Documenter, Symbol("##3#8")), ::String) at ./file.jl:96
 [6] #makedocs#1(::Bool, ::Documenter.Writers.HTMLWriter.HTML, ::Nothing, ::Nothing, ::Nothing, ::Nothing, ::Base.Iterators.Pairs{Symbol,Any,NTuple{13,Symbol},NamedTuple{(:build, :modules, :clean, :doctest, :linkcheck, :linkcheck_ignore, :strict, :checkdocs, :sitename, :authors, :analytics, :pages, :assets),Tuple{String,Array{Module,1},Bool,Bool,Bool,Array{String,1},Bool,Symbol,String,String,String,Array{Any,1},Array{String,1}}}}, ::Function) at /home/ederag/share/prog/julia/dev/julia-dev/doc/deps/packages/Documenter/zOHWz/src/Documenter.jl:266
 [7] (::getfield(Documenter, Symbol("#kw##makedocs")))(::NamedTuple{(:build, :modules, :clean, :doctest, :linkcheck, :linkcheck_ignore, :strict, :checkdocs, :format, :sitename, :authors, :analytics, :pages, :assets),Tuple{String,Array{Module,1},Bool,Bool,Bool,Array{String,1},Bool,Symbol,Documenter.Writers.HTMLWriter.HTML,String,String,String,Array{Any,1},Array{String,1}}}, ::typeof(makedocs)) at ./none:0
 [8] top-level scope at none:0
 [9] include at ./boot.jl:326 [inlined]
 [10] include_relative(::Module, ::String) at ./loading.jl:1038
 [11] include(::Module, ::String) at ./sysimg.jl:29
 [12] exec_options(::Base.JLOptions) at ./client.jl:267
 [13] _start() at ./client.jl:436
in expression starting at /home/ederag/share/prog/julia/dev/julia-dev/doc/make.jl:168
make[1]: *** [Makefile:39: html] Error 1
make: *** [Makefile:91: docs] Error 2
```
Sorry, I'm giving up.
Let someone familiar with the docs building process take it from here.

current commit: 674a50450dfe33968808c5056462eaac038e3f79 (HEAD, upstream/pr/30556, origin/patch-1)
parent commit: 2c6422041c8b903b7efb843db881434bb2bc43c4

Reply:
OK, second try, found a [way to really wipe clean the repository](https://discourse.julialang.org/t/julia-1-0-3-testing-period/18610/25).
The link works.
Please squash merge; first commit has no interest.
Thanks for the review.